samza-api/src/main/java/org/apache/samza/metadatastore/MetadataStore.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.metadatastore;
+
+import org.apache.samza.container.TaskName;
+import org.apache.samza.runtime.LocationId;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Store abstraction responsible for managing the metadata of a samza job and is agnostic of the
+ * deployment model(yarn/standalone) of the samza job. Serialization and deserialization of the metadata should be
+ * hidden within the implementation.
```

Comment:
Re: serde should be within the implementation: that's implied by the return types in the interface, don't need to mention here.